Applications/SlicerApp/Testing/Python/WebEngine.py

```diff
@@ -0,0 +1,230 @@
+  """
+
+  def __init__(self, parent):
+    ScriptedLoadableModuleWidget.__init__(self, parent)
+    self.webWidgets = [] # hold references so windows persist
+
+  def setup(self):
+    ScriptedLoadableModuleWidget.setup(self)
+    # Instantiate and connect widgets ...
+
+    # Collapsible button
+    sitesCollapsibleButton = ctk.ctkCollapsibleButton()
+    sitesCollapsibleButton.text = "Sample Sites"
+    self.layout.addWidget(sitesCollapsibleButton)
+
+    # Layout within the collapsible button
+    sitesFormLayout = qt.QFormLayout(sitesCollapsibleButton)
+
+    # site buttons
+    buttons = []
+    self.sites = [
+      {
+        "label": "Web Console", "url": "http://localhost:1337"
+      },
+      {
+        "label": "Crowds Cure Cancer", "url": "http://cancer.crowds-cure.org"
+      },
+      {
+        "label": "Slicer Home Page", "url": "http://slicer.org"
+      },
+      {
+        "label": "MorphoSource", "url": "https://www.morphosource.org"
+      },
+      {
+        "label": "Slicer SampleData", "url": "https://www.slicer.org/wiki/SampleData"
+      },
+      {
+        "label": "SlicerMorph", "url": "https://slicermorph.github.io"
+      },
+    ]
+    for site in self.sites:
+      button = qt.QPushButton(site["label"])
+      button.toolTip = "Open %s" % site["url"]
+      sitesFormLayout.addWidget(button)
+      onClick = lambda click, site=site: self.onSiteButtonClicked(site)
+      button.connect('clicked(bool)', onClick)
+      buttons.append(button)
+
+    button = qt.QPushButton("Close All")
+    button.toolTip = "Close all the web views"
+    button.connect('clicked(bool)', self.onCloseAll)
+    self.layout.addWidget(button)
+
+    # Add vertical spacer
+    self.layout.addStretch(1)
+
+  def onSiteButtonClicked(self, site):
+    webWidget = slicer.qSlicerWebWidget()
+    slicerGeometry = slicer.util.mainWindow().geometry
+    webWidget.size = qt.QSize(1536,1024)
+    webWidget.pos = qt.QPoint(slicerGeometry.x() + 256, slicerGeometry.y() + 128)
+    webView = webWidget.webView()
+    webView.url = qt.QUrl(site["url"])
+    webWidget.show()
+    self.webWidgets.append(webWidget)
+
+  def onCloseAll(self):
+    for widget in self.webWidgets:
+      del widget
+    self.webWidgets = []
+
+
+class WebEngineTest(ScriptedLoadableModuleTest):
+  """
+  This is the test case for your scripted module.
+  Uses ScriptedLoadableModuleTest base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def setUp(self):
+    """ Do whatever is needed to reset the state - typically a scene clear will be enough.
+    """
+    self.gotResponse = False
+
+  def runTest(self):
+    """Run as few or as many tests as needed here.
+    """
+    self.setUp()
+    self.test_WebEngine1()
+
+  def onEvalResult(self, js, result):
+    if js == "valueFromSlicer;":
+      if result != "42":
+        Exception("Did not get back expected result!")
```

Comment:
In python, it is usually better to return more specific exception. What about using `RuntimeException`  or do `assert result == "42"`